### ISSUE TITLE
Embedded JS in Alpine x-data

### ISSUE THREAD

**Issue by cheesegrits** [Stillat/blade-parser-typescript#55]
Any chance we can get embedded JS in Alpine x-data strings formatted?

Big ask, I know, but this is one of my biggest headaches with Blade formatting.

Thanks for all the hard work on this.  It's awesome having someone show Blade formatting some serious love.

```html
    <div
        x-ignore
        ax-load
        ax-load-src="{{ \Filament\Support\Facades\FilamentAsset::getAlpineComponentSrc('file-upload', 'filament/forms') }}"
        x-data="fileUploadFormComponent({
            acceptedFileTypes: @js($getAcceptedFileTypes()),
            deleteUploadedFileUsing: async (fileKey) => {
                return await $wire.deleteUploadedFile(@js($statePath), fileKey)
            },
            getUploadedFilesUsing: async () => {
                return await $wire.getFormUploadedFiles(@js($statePath))
            },
            uploadUsing: (fileKey, file, success, error, progress) => {
                $wire.upload(`{{ $statePath }}.${fileKey}`, file, () => {
                    success(fileKey)
                }, error, progress)
            },
        })"
        wire:ignore
    >
```

**Comment by zepfietje** [comment]
Just for context, @cheesegrits brought this up with me and I told him to create an issue for it, @JohnathonKoster. I'd also already been thinking about this for the Filament codebase. Would be nice if this worked in all Alpine directives (like `x-text`, `x-show`).

**Comment by JohnathonKoster** [comment]
This is definitely something I can look into. Will update here as I know more 🙂

**Comment by zepfietje** [comment]
Awesome!

**Comment by JohnathonKoster** [comment]
WIP todo-list (to be updated as review/prototype work continues):

- [ ] Refactor fragmented-HTML parser to account for attributes like `<input x-on:change{{ $type === 'number' ? '.debounce.1s' : null }}="..." />`
- [ ] Handle case when attribute documents contain simple content like `{}`
- [ ] Refactor nested JavaScript document process to be a pre-processor (similar to CSS class strings) to continue leveraging the current Pint process
- [ ] Return original attribute content when it contains Directive pairs, conditions, etc. **

** Original content will be returned to prevent layout changes from impacting generated JavaScript, as there is no reliable way to predict what the final output will be in all situations. Simple directives, echos, etc. will be fine.

**Comment by JohnathonKoster** [comment]
This is now available starting with v1.6 🎉 - there are some configuration options available. I've added them in this comment for convenience, but this information is also available within the readme:

## Formatting JavaScript Inside Attributes

The Blade formatter will format JavaScript inside Alpine.js directives by default (starting with 1.6.0). This can be disabled by adding a `formatJsAttributes` configuration item to your `.blade.format.json`:

```json
{
    "formatJsAttributes": false,
}
```

If you would like to exclude a list of attributes from being formatted, you may add a list of regular expressions to the `.blade.format.json` file. By default, it contains the following items:

```json
{
    "formatJsAttributes": true,
    "excludeJsAttributes": [
        "^v-"
    ]
}
```

If you'd like to expand the list of attributes to be formatted, you may add a list of regular expressions to the `.blade.format.json` file. By default this list contains the following items:

```json
{
    "formatJsAttributes": true,
    "includeJsAttributes": [
        "^x-",
        "^ax-"
    ]
}
```

### Modifying JavaScript Prettier Options

You can change which prettier options are applied to attribute content by adding a `attributeJsOptions` configuration object to your `.blade.format.json` file. For example, to increase the print width and disable the insertion of semicolons, you could add the following:

```json
{
    "attributeJsOptions": {
        "semi": false,
        "printWidth": 120
    }
}
```

**Comment by zepfietje** [comment]
Great work, @JohnathonKoster! I'll give it a go soon.
